src/vm/analysis/type_checker/natives/sequences.rs

```diff
@@ -391,3 +391,39 @@ pub fn check_special_index_of(
 
     TypeSignature::new_option(TypeSignature::UIntType).map_err(|e| e.into())
 }
+
+pub fn check_special_slice(
+    checker: &mut TypeChecker,
+    args: &[SymbolicExpression],
+    context: &TypingContext,
+) -> TypeResult {
+    check_argument_count(3, args)?;
+
+    // Check sequence
+    let seq_type = checker.type_check(&args[0], context)?;
+    let seq = match &seq_type {
+        TypeSignature::SequenceType(seq) => TypeSignature::SequenceType(seq.clone()),
+        _ => return Err(CheckErrors::ExpectedSequence(seq_type.clone()).into()),
+    };
+    // TODO: runtime_cost
+
+    // Check position argument
+    let position = checker.type_check(&args[1], context)?;
+    match position {
+        TypeSignature::UIntType => Ok(()),
+        _ => Err(CheckErrors::TypeError(TypeSignature::UIntType, position)),
+    }?;
+    // TODO: runtime_cost
+
+    // Check length argument
+    let length = checker.type_check(&args[2], context)?;
+    match length {
+        TypeSignature::UIntType => Ok(()),
+        _ => Err(CheckErrors::TypeError(TypeSignature::UIntType, length)),
+    }?;
```

Comment:
I think both of these can use `checker.type_check_expects` to simplify the check:

```suggestion
    let position = checker.type_check_expects(&args[1], context, &TypeSignature::UIntType)?;
    // TODO: runtime_cost

    // Check length argument
    let length = checker.type_check_expects(&args[2], context, &TypeSignature::UIntType)?;
```